python/shotgun_desktop/settings.py

```diff
@@ -160,3 +160,22 @@ def _get_value(self, section, key, type_cast=str, default=None):
             return default
         else:
             return type_cast(self._global_config.get(section, key))
+
+    def _get_filtered_proxy(self):
+        """
+        :returns: Returns the proxy settings with credentials masked.
+        """
+        # If there is an address available
+        proxy = self.default_http_proxy
+
+        # If the proxy is not set, return the default value.
+        if not proxy:
+            return proxy
+
+        # If there is a username and password in the proxy string
+        if "@" in proxy:
+            # Filter out the username and password
+            # Given xzy:123@localhost or xyz:12@3@locahost, this will return localhost in both cases
+            return "xxx@%s" % proxy.split("@")[-1]
+        else:
```

Comment:
minor suggestion. you could change the second conditional to be:

```
if proxy and "@" in proxy:
```

which would allow you to remove the first conditional and the `else` in the second conditional and then just `return proxy` at the end of the method.